Assets/Stack mini game/Scripts/StackMinigameUI.cs

```diff
@@ -0,0 +1,35 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.UI;
+
+public class StackMinigameUI : MonoBehaviour
+{
+    [SerializeField] private GameObject stackButton;
+    [SerializeField] private GameObject replayButton;
+
+    [SerializeField] private Text outcomeText;
+
+
+    public void SetLoseUI()

```

Comment:
`  public void SetEndOfRoundUI(bool hasWon)
    {
        outcomeText.text = hasWon ? "You won!" : "Game over!";

        stackButton.SetActive(false);
        replayButton.SetActive(true);
    }

    public void SetNewGameUI()
    {
        outcomeText.text = "";
        stackButton.SetActive(true);
        replayButton.SetActive(false);
    }`

Reply:
Maybe we could go with this. I need to reset the outcomeText to "" between each game.
Moreover, once we move this code to the Starlight Express, we'll need to reset the UI OnEnable when the player starts the mini-game and between each play session. We can't have a message showing "You won" when the player starts a new game.

Reply:
At the beginning, I had only one function with a very open signature. You could pass a string for the outcomeText and a bool for the state of the button. I didn't like that since it added complexity to the system, since any user of the class could basically change the outcomeText.

But I think your solution is better. The signature is still open but it's more limited, and there's less possibility to make a mistake.